src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -349,17 +332,39 @@ public String getPublishArtifactStage(ArtifactPublishingConfig config,
 
         StringBuilder snippet = new StringBuilder("stage('Publish Artifact') {\n");
 
-        snippet.append("\tsteps {\n");
-        snippet.append("\t\t" + "withCredentials([file(credentialsId: '").append(config.getCredentialId()).append("', variable: 'FILE')]) {\n");
+        snippet.append("steps {\n");
+        snippet.append("" + "withCredentials([file(credentialsId: '").append(config.getCredentialId()).append("', variable: 'FILE')]) {\n");
 
         for(HashMap<String, String> artifact: publishArtifacts){
-            snippet.append("\t\t\tsh 'scp -i $FILE ").append(artifact.get("from")).append(" ").append(config.getUser()).append("@").append(config.getHost()).append(":").append(artifact.get("to")).append("'\n");
+            snippet.append("sh 'scp -i $FILE ").append(artifact.get("from")).append(" ").append(config.getUser()).append("@").append(config.getHost()).append(":").append(artifact.get("to")).append("'\n");
         }
 
-        snippet.append("\t\t}\n");
-        snippet.append("\t}\n");
+        snippet.append("}\n");
+        snippet.append("}\n");
         snippet.append("}\n");
 
         return snippet.toString();
     }
+
+    public StringBuilder autoAddTabs(StringBuilder pipelineScript){
+        int numOfTabs = 0;
+
+        for(int i = 0; i < pipelineScript.length(); i++){
+            if(pipelineScript.charAt(i) == '{'){
+                numOfTabs++;
+            }
+
+            if(i + 1 != pipelineScript.length() &&
+                    pipelineScript.charAt(i) == '\n' &&
+                    pipelineScript.charAt(i + 1) == '}'){
+                numOfTabs--;
+            }
+
+            if(pipelineScript.charAt(i) == '\n'){
+                pipelineScript.insert(i + 1, StringUtils.repeat("\t", numOfTabs));
```

Comment:
This changes the the length of the `pipelineScript`, which means the value of the conditional on line 352 changes dynamically. It might work for certain cases... but it is confusing logic. Probably hard to maintain in the future. I would suggest a different approach. I would create the entire script as a list of lines without indentation and without the `\n` at the end of each line. After the entire script is created as a list of Strings, I would have another method looping through the list. Inside the loop I would track the indentation level and I would concatenate the indentation string to the list element string, and append the result to the return value of the method.

Something like:
 ```
// pseudo-code
String indentScript(String[] scriptLines) {
    String indentedScript = "";
    int indent = 0;
    for (String line : scriptLines) {
        indentedScript += (indent * "\t") + line + System.getProperty("line.separator");
        if ("{" in line) then indent++
        if ("}" in line) then indent--
    }
    return indentedString;
}
```
Of course the method could return a list of Strings and take in a list of Strings, or any other suitable data type, it is an optimization that is left as an exercise to the coder :-)

Reply:
This is a good point made by Martin about the thing your looping through changing length, and should be fixed before this PR is merged.

As for the pseudo code example, to me that looks like a good solution, but up to you to decide.